File: Type-3/JoseVitorSAmorim_3.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>

void imprimirMoeda(const char* texto, float valor) {
    char buffer[50];
    sprintf(buffer, "%.2f", valor); 
    
    for(int i = 0; buffer[i] != '\0'; i++) {
        if(buffer[i] == '.') {
            buffer[i] = ',';
        }
    }
    printf("%s%s\n", texto, buffer);
}

int main() {
    float valor_original = 0, desconto = 0, valor_montante = 0, desconto_vip = 0;
    char vip;

    while (1) {
        printf("Valor da compra: R$");
        scanf("%f", &valor_original);

        printf("Cliente VIP[S/N]: ");
        scanf(" %c", &vip);
        vip = toupper(vip);

        if (valor_original >= 500) {
            desconto = valor_original * 0.1;
            valor_montante = valor_original - desconto;
            if (vip == 'S') {
                desconto_vip = valor_montante * 0.05;
                valor_montante -= desconto_vip;
            }
            break;
        } 
        else if (valor_original < 500 && valor_original > 0) {
            desconto = valor_original * 0.05;
            valor_montante = valor_original - desconto;
            if (vip == 'S') {
                desconto_vip = valor_montante * 0.05;
                valor_montante -= desconto_vip;
            }
            break;
        } 
        else {
            printf("\033[31mCalma ai patrao, tem algo errado\033[m\n");
            printf("\033[33mTente novamente\033[m\n");
        }
    }

    printf("\n---Recibo de compra---\n");
    
    imprimirMoeda("Valor original: R$", valor_original);
    
    if (vip == 'S') {
        imprimirMoeda("Desconto: R$", desconto + desconto_vip);
    } 
    else {
        imprimirMoeda("Desconto: R$", desconto);
    }
    
    imprimirMoeda("Valor final: R$", valor_montante);
    int tamanho = strlen("---Recibo de compra---");
    for(int i = 0; i < tamanho; i++) {
        printf("-");
    }
    printf("\n");

    return 0;
}

File: Type-3/JoseVitorSAmorim_3.java
import java.util.Scanner;
public class JoseVitorSAmorim_3 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        float valorOriginal = 0, desconto = 0, valorMontante = 0, descontoVip = 0;
        String vip = "";
        while (true) {
            System.out.print("Valor da compra: R$");
            valorOriginal = scanner.nextFloat();
            System.out.print("Cliente VIP[S/N]: ");
            vip = scanner.next().toUpperCase();
            if (valorOriginal >= 500) {
                desconto = (float) (valorOriginal * 0.1);
                valorMontante = valorOriginal - desconto;
                if (vip.equals("S")) {
                    descontoVip = (float) (valorMontante * 0.05);
                    valorMontante -= descontoVip;
                }
                break;
            } 
            else if (valorOriginal < 500 && valorOriginal > 0) {
                desconto = (float) (valorOriginal * 0.05);
                valorMontante = valorOriginal - desconto;
                if (vip.equals("S")) {
                    descontoVip = (float) (valorMontante * 0.05);
                    valorMontante -= descontoVip;
                }
                break;
            } 
            else {
                System.out.println("\033[31mCalma ai patrão, tem algo errado\033[m");
                System.out.println("\033[33mTente novamente\033[m");
            }
        }
        System.out.println("\n---Recibo de compra---");
        System.out.println("Valor original: R$" + String.format("%.2f", valorOriginal).replace(".", ","));
        if (vip.equals("S")) {
            System.out.println("Desconto: R$" + String.format("%.2f", desconto + descontoVip).replace(".", ","));
        } 
        else {
            System.out.println("Desconto: R$" + String.format("%.2f", desconto).replace(".", ","));
        }
        System.out.println("Valor final: R$" + String.format("%.2f", valorMontante).replace(".", ","));
        System.out.println("-".repeat("---Recibo de compra---".length()));
        scanner.close();
    }
}
